src/js/Calculator.js

```diff
@@ -0,0 +1,82 @@
+import { STRATEGY } from './const/index.js';
+import DigitStrategy from './strategies/DigitStrategy.js';
+import ModifierStrategy from './strategies/ModifierStrategy.js';
+import OperationStrategy from './strategies/OperationStrategy.js';
+import { isEmptyArray, isNull } from './utils/common.js';
+import { beforeunload } from './utils/domApi.js';
+
+export const defaultState = {
+  x: null,
+  operator: null,
+  y: null,
+};
+
+class Calculator {
+  #$calculator;
+  #$total;
+
+  constructor({ $calculator, $total }) {
+    this.#$calculator = $calculator;
+    this.#$total = $total;
+
+    this.#init();
+  }
+
+  #init() {
+    this.#render(defaultState);
+    this.#bindEvent();
+    this.#removeEvent();
+  }
+
+  #render(state) {
+    this.#mutateState(state);
+    this.#$total.innerText = this.#convertToDisplayValue();
+  }
+
+  #mutateState(newState) {
+    this.state = newState ? { ...newState } : this.state;
+  }
+
+  #convertToDisplayValue() {
+    const values = Object.values(this.state);
+    const displayValues = values.filter((v) => !isNull(v));
+    if (isEmptyArray(displayValues)) return '0';
+    return displayValues.join('');
+  }
+
+  #bindEvent() {
+    this.#$calculator.addEventListener('click', this.#handleClick.bind(this));
+  }
+
+  #removeEvent() {
+    beforeunload(() =>
+      this.#$calculator.removeEventListener(
+        'click',
+        this.#handleClick.bind(this)
+      )
+    );
+  }
+
+  #handleClick({ target }) {
+    const strategy = this.#strategyCreator(target.dataset);
+
+    try {
+      const newState = strategy?.mutateState(target, this.state);
+      this.#render(newState);
```

Comment:
```suggestion
      this.#mutateState(strategy?.mutateState(target, this.state));
      this.#render();
```

이렇게 해봤으면 어땠을까요?
이러면 이제 render는 state 변경을 알 필요 없이 dom 변경만 잘 관리하면 되겠네요~

Reply:
🤩🤩🤩🤩🤩 왜 이 생각을 못 했을까요!!!!!!! 객체로 수정하면서 반영했습니다!! 감사합니다 👍👍👍